### PR TITLE
test(share_plus): Integration test

### PR DESCRIPTION
**Pull request by Chever-John**
## Description

Now our integration test supports the Android platform.
Write an integration test of the share_plus plugin supporting the Android Platform.

## Related Issues

This is a GSoC task, so there are no associated issues for now.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated the version in `pubspec.yaml` and `CHANGELOG.md`.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: packages/share_plus/share_plus/example/integration_test/share_plus_test.dart [hunk 1]
```diff
@@ -14,4 +14,8 @@ void main() {
   testWidgets('Can launch share', (WidgetTester tester) async {
     expect(Share.share('message', subject: 'title'), completes);
   });
+
+  testWidgets('Test', (WidgetTester tester) async {
```
Comment (miquelbeltran) [inline packages/share_plus/share_plus/example/integration_test/share_plus_test.dart:18]: Can you find a better test name? `Test` is not good.

`Can launch shareWithResult` for example

**Review by miquelbeltran — CHANGES_REQUESTED** [review]

File: packages/share_plus/share_plus/example/integration_test/share_plus_test.dart (hunk 1 above)
Comment (Chever-John) [reply packages/share_plus/share_plus/example/integration_test/share_plus_test.dart:18]: Sure

File: packages/share_plus/share_plus/example/integration_test/share_plus_test.dart [hunk 2]
```diff
@@ -14,4 +14,9 @@ void main() {
   testWidgets('Can launch share', (WidgetTester tester) async {
     expect(Share.share('message', subject: 'title'), completes);
   });
+
+  testWidgets('Tests if the expected non-null result is returned',
```
Comment (miquelbeltran) [inline packages/share_plus/share_plus/example/integration_test/share_plus_test.dart:18]: I mean, it is an improvement :) but could you add a reference to the method name shareWithResult?

e.g. "Can launch shareWithResult and result is not null."

Comment (Chever-John) [reply packages/share_plus/share_plus/example/integration_test/share_plus_test.dart:18]: Yes!!!

**Comment by miquelbeltran** [comment]
I think you misunderstood me. 

The test descriptions that you set are not valid and must be changed.

Please set the following test descriptions:

- "Can launch share"
- "Can launch shareWithResult"

**Comment by Chever-John** [comment]
> 

Sorry, I thought it was to improve my vocabulary.

But it is true that the naming you gave would be a bit more exact. Learned.

So let's see if the latest submission meets the requirements.

**Review by miquelbeltran — APPROVED** [review]

**Pull request merged**